metagraph/src/graph/representation/succinct/boss_chunk_construct.cpp

```diff
@@ -347,22 +348,30 @@ using Decoder = common::EliasFanoDecoder<T>;
 template <typename T_REAL>
 std::vector<std::string> split(size_t k,
                                const std::filesystem::path &dir,
-                               const ChunkedWaitQueue<T_REAL> &kmers) {
+                               const ChunkedWaitQueue<T_REAL> &kmers,
+                               BuildCheckpoint* checkpoint) {
     using T_INT_REAL = get_int_t<T_REAL>;
 
     const uint8_t alphabet_size = KmerExtractor2Bit().alphabet.size();
 
     size_t chunk_count = std::pow(alphabet_size, 2);
 
-    logger->trace("Splitting k-mers into {} chunks...", chunk_count);
-
     std::vector<Encoder<T_INT_REAL>> sinks;
     std::vector<std::string> names(chunk_count);
     for (size_t i = 0; i < names.size(); ++i) {
         names[i] = dir/("real_F_W_" + std::to_string(i));
+    }
+
+    if (checkpoint->checkpoint() > 2) {
+        logger->info("Skipping splitting k-mers into chunks");
+        return names;
+    }
```

Comment:
I got confused here. Please print that the k-mers have already been split into chunks in the last checkpoint.
Otherwise the logging message is misleading.

Reply:
also, add `assert(checkpoint->checkpoint() >= 2)` in the beginning of the function. It can't be 1, right? Similar thing for the other functions with next checkpoints.